src/pilot/templates/post-install.yaml

```diff
@@ -1,60 +1,60 @@
-# limitations under the License.
-
-heat_template_version: 2014-10-16
-
-description: >
-  Example extra config for post-deployment
-
-parameters:
-  servers:
-    type: json
-
-resources:
-  ExtraDeployments:
-    type: OS::Heat::StructuredDeployments
-    properties:
-      servers:  {get_param: servers}
-      config: {get_resource: ExtraConfig}
-      # Do this on CREATE/UPDATE (which is actually the default)
-      actions: ['CREATE', 'UPDATE']
-
-  ExtraConfig:
-    type: OS::Heat::SoftwareConfig
-    properties:
-      group: script
-      config: |
-        #!/bin/bash
-
-        set -x
-        if [[ $HOSTNAME =~ "compute" ]] ; then
-          VCPUPINSET=$(crudini --get /root/parameters.ini DEFAULT novacpus)
-          crudini --set /etc/nova/nova.conf DEFAULT vcpu_pin_set $VCPUPINSET
-          systemctl restart openstack-nova-compute.service
-          tuned_service=/usr/lib/systemd/system/tuned.service
-          grep -q "network.target" $tuned_service
-          if [ "$?" -eq 0 ]; then
-            sed -i '/After=.*/s/network.target//g' $tuned_service
-          fi
-          grep -q "Before=.*network.target" $tuned_service
-          if [ ! "$?" -eq 0 ]; then
-            grep -q "Before=.*" $tuned_service
-            if [ "$?" -eq 0 ]; then
-              sed -i 's/^\(Before=.*\)/\1 network.target openvswitch.service/g' $tuned_service
-            else
-              sed -i '/After/i Before=network.target openvswitch.service' $tuned_service
-            fi
-          fi
-          systemctl daemon-reload
-        fi
+# Copyright (c) 2016-2018 Dell Inc. or its subsidiaries.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+heat_template_version: 2014-10-16
+
+description: >
+  Example extra config for post-deployment
+
+parameters:
+  servers:
+    type: json
+
+resources:
+  ExtraDeployments:
+    type: OS::Heat::StructuredDeployments
+    properties:
+      servers:  {get_param: servers}
+      config: {get_resource: ExtraConfig}
+      # Do this on CREATE/UPDATE (which is actually the default)
+      actions: ['CREATE', 'UPDATE']
+
+  ExtraConfig:
+    type: OS::Heat::SoftwareConfig
+    properties:
+      group: script
+      config: |
+        #!/bin/bash
+
+        set -x
+        if [[ $HOSTNAME =~ "compute" ]] ; then
+          VCPUPINSET=$(crudini --get /root/parameters.ini DEFAULT novacpus)
+          crudini --set /etc/nova/nova.conf DEFAULT vcpu_pin_set $VCPUPINSET
```

Comment:
Why doing this? Is the NovaVcpuPinSet not already set to the appropriate values? If so, then the nova.conf will be already populated.

Reply:
We can't set whole set of vcpus to in nova.conf as we need to subtract pmd cpus from vcpus and rest can go to nova.conf .
Refer -
[DEFAULT]
hostcpus = 0,24,2,26,1,25,3,27
pmdcpus = 5,29,4,28,6,30,8,32,7,31,9,33
novacpus = 10,34,12,36,14,38,16,40,18,42,20,44,22,46,11,35,13,37,15,39,17,41,19,43,21,45,23,47
isolcpus = 5,29,4,28,6,30,8,32,7,31,9,33,10,34,12,36,14,38,16,40,18,42,20,44,22,46,11,35,13,37,15,39,17,41,19,43,21,45,23,47
grub = default_hugepagesz=1GB hugepagesz=1G hugepages=116 iommu=pt intel_iommu=on

Please note that there are no changes to post-install.yaml (same as master's)

Reply:
I see, but NovaVcpuPinSet is handled by the appropriate TripleO Paramater and therefore should managed anywhere else. So it's good to substract cpus, but once you have the remaining vcpus then you should update the line that goes along with the dell-environment.yaml:
# NovaVcpuPinSet: ['8-47']

Reply:
We can apply NovaVcpuPinSet in environment and we are calculating pmdcpus in first-boot-ovs-dpdk.yaml so not possible to set that to NovaVcpuPinSet. 
We are applying the vcpus to NovaVcpuPinSet when dpdk is not enable.

Reply:
https://github.com/uday344/JetPack/blob/3b13861d223ac8c1179f6834bb25d1996a10f0aa/src/pilot/dell_nfv.py#L273